lib/tasks/import_prebakedstats.rake

```diff
@@ -15,6 +15,10 @@ namespace :import do
         import_new_csv_file(habitat.name, csv_file)
       end
     end
+
+    # Import occurrences for any countries that are missing (disregarding regions for the moment)
+    # using habitat_country_presence.csv
```

Comment:
I'm not sure I understand why there are missing occurrences/why running whichever importer usually imports these values wouldn't fix this.

Reply:
This was a little tricky to solve, but I pinpointed it to the fact that there is missing species data for certain countries in the Redlist CSV (as in there are no rows in that CSV with the ISO3s of those countries, hence no species data), but which have habitat coverage data present (which is provided by `habitat_country_coverage` which has the full list of countries/territories). 

As we only get a portion of the full statistics in each CSV/set of CSVs, I decided to make this part of the 'prebaked' stats task since it directly deals with creating and updating GeoEntityStats, which are the crucial model that the statistics rely on to populate country pages. In this case, after the main import is run with the CSVs in the `habitat_coverage_protection` folder, I then use another CSV, the `habitat_presence_country` CSV, which contains all of the countries, plus their occurrences. *But*, there is no data on the coverage and protected area for each habitat in this CSV, it only exists in the previous set of CSVs. Hence I have to use both sources to fully populate the statistics. Basically, I fetch any GeoEntityStats with a value of `nil` for the `occurrence` attribute to trace the parent country, grab the ISO3 of the country and then use the `occurrence` values in `habitat_presence_country` to update each habitat in turn. 

Hope that makes sense

Reply:
To be honest, it's better that all of this information is combined into one CSV, but that isn't the case

Reply:
I think I follow! Just to check - if you run db:seed on a refresh db, will this import everything?

Reply:
Yes it should